python/beeai_framework/tools/tool.py

```diff
@@ -96,7 +110,19 @@ async def executor(_: RetryableContext) -> TOutput:
                     nonlocal error_propagated
                     error_propagated = False
                     await context.emitter.emit("start", ToolStartEvent(input=validated_input, options=options))
-                    return await self._run(validated_input, options, context)
+
+                    if self.cache.enabled:
+                        cache_key = self._generate_key(input, options)
+                        if await self.cache.has(cache_key):
+                            result = await self.cache.get(cache_key)
+                            if result:
+                                return result
```

Comment:
I would just do `await self.cache.get(cache_key)`, and if the value isn't `None`, I would return it.

Reply:
7f8c530f16ed2a9bf77b57895be496739f078164